include/fmt/format-inl.h

```diff
@@ -18,7 +18,7 @@
 #  include <locale>
 #endif
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(FMT_WINDOWS_NO_WCHAR)
```

Comment:
Can this be detected automatically e.g. via some predefined macro or `__has_include(<io.h>)`?

Reply:
Done. It can't always be detected automatically but I've added the 2 cases where it can (missing `<io.h>` on xbox and defined WINVER on properly configured Windows projects). Even when it can't be detected automatically, one can always `add_definitions(FMT_WINDOWS_NO_WCHAR)` in the fmt subproject, so that's not a problem.

Reply:
I was hoping that it will be less messy =). It is particularly weird to have this tested in core.h which explicitly avoids `wchar_t`. Let's go back to just requiring the user to define `FMT_WINDOWS_NO_WCHAR` (and without the config option).

Reply:
Done!

![image](https://github.com/diasurgical/devilutionX/assets/216339/58ac8947-cd3d-49b0-a185-90cdb96e512f)